daemon/algod/api/server/v1/handlers/handlers.go

```diff
@@ -216,17 +217,74 @@ func assetFreezeTxEncode(tx transactions.Transaction, ad transactions.ApplyData)
 	}
 }
 
-func txWithStatusEncode(tr node.TxnWithStatus) (v1.Transaction, error) {
+func txWithStatusEncode(tr node.TxnWithStatus, l *data.Ledger) (v1.Transaction, error) {
 	s, err := txEncode(tr.Txn.Txn, tr.ApplyData)
 	if err != nil {
 		err = decorateUnknownTransactionTypeError(err, tr)
 		return v1.Transaction{}, err
 	}
 	s.ConfirmedRound = uint64(tr.ConfirmedRound)
 	s.PoolError = tr.PoolError
+	s.TransactionResults = &v1.TransactionResults{
+		CreatedAssetIndex: computeAssetIndexFromTxn(tr, l),
+	}
```

Comment:
Seems like we're enforcing behavior for future callers. i.e. all future functions must return the empty value in order to be omitted. I believe that the addition of extra information should happen at the caller level.

Reply:
fixed :+1: